### ISSUE TITLE
Provide validation based on an xsd

### ISSUE THREAD

**Issue by apupier** [angelozerr/xml-languageserver#4]

**Comment by angelozerr** [comment]
Indeed.  https://github.com/microclimate-devops/xml-language-server/tree/master/server/xml-server seems doing that.

**Comment by angelozerr** [comment]
It starts working, but a lot of features to implement again like
 
 * manage better start/end location for error (because XSD validation from xerces manages just a start position and not a range)
 * cache XML Schema, because for the moment, the XML Schema is loaded each time you type something. But there is not freeze because publish diagnostics is done in async mode. To fix that, I think XML catalog should help us and I need to read https://xerces.apache.org/xerces2-j/faq-grammars.html
